c_glib/test/test-csv-writer.rb

```diff
@@ -0,0 +1,190 @@
+
+  def test_write_record_batch
+    message_data = ["Start", "Shutdown"]
+    count_data = [2, 9]
+    message_field = Arrow::Field.new("message", Arrow::StringDataType.new)
+    count_field = Arrow::Field.new("count", Arrow::Int64DataType.new)
+    schema = Arrow::Schema.new([message_field, count_field])
+
+    buffer = Arrow::ResizableBuffer.new(0)
+    output = Arrow::BufferOutputStream.new(buffer)
+    begin
+      csv_writer = Arrow::CSVWriter.new(output, schema, nil)
+      begin
+        record_batch = Arrow::RecordBatch.new(schema,
+                                              message_data.size,
+                                              [
+                                                build_string_array(message_data),
+                                                build_int64_array(count_data),
+                                              ])
+        csv_writer.write_record_batch(record_batch)
+      ensure
+        csv_writer.close
+        assert do
+          csv_writer.closed?
+        end
+      end
+    ensure
+      output.close
+    end
+
+    csv_output = buffer.data.to_s
+    expected = <<~CSV
+      "message","count"
+      "Start",2
+      "Shutdown",9
+    CSV
+    assert_equal(expected, csv_output)
+  end
+
+  def test_write_table
+    message_data = ["Start", "Shutdown", "Reboot"]
+    count_data = [2, 9, 5]
+    message_field = Arrow::Field.new("message", Arrow::StringDataType.new)
+    count_field = Arrow::Field.new("count", Arrow::Int64DataType.new)
+    schema = Arrow::Schema.new([message_field, count_field])
+
+    buffer = Arrow::ResizableBuffer.new(0)
+    output = Arrow::BufferOutputStream.new(buffer)
+    begin
+      csv_writer = Arrow::CSVWriter.new(output, schema, nil)
+      begin
+        table = Arrow::Table.new(schema,
+                                 [
+                                   build_string_array(message_data),
+                                   build_int64_array(count_data),
+                                 ])
+        csv_writer.write_table(table)
+      ensure
+        csv_writer.close
+        assert do
+          csv_writer.closed?
+        end
+      end
+    ensure
+      output.close
+    end
+
+    csv_output = buffer.data.to_s
+    expected = <<~CSV
+      "message","count"
+      "Start",2
+      "Shutdown",9
+      "Reboot",5
+    CSV
+    assert_equal(expected, csv_output)
+  end
+
+
+  sub_test_case("options") do
+    def setup
+      @options = Arrow::CSVWriteOptions.new
+    end
+
+    def test_include_header
+      assert do
+        @options.include_header?
+      end
+      @options.include_header = false
+      assert do
+        not @options.include_header?
+      end
+    end
+
+    def test_batch_size
+      assert_equal(1024, @options.batch_size)
+      @options.batch_size = 2048
+      assert_equal(2048, @options.batch_size)
+    end
+
+    def test_delimiter
+      assert_equal(44, @options.delimiter) # 44 is the ASCII code for comma
```

Comment:
How about using descriptive code instead of comment?

```suggestion
      assert_equal(".".ord, @options.delimiter)
```